main/src/ca/uwaterloo/flix/language/dbg/AstPrinter.scala

```diff
@@ -0,0 +1,55 @@
+package ca.uwaterloo.flix.language.dbg
+
+import ca.uwaterloo.flix.api.Flix
+import ca.uwaterloo.flix.language.ast.SourceLocation
+import ca.uwaterloo.flix.language.dbg.Doc.Indent
+import ca.uwaterloo.flix.util.InternalCompilerException
+
+import java.nio.file.{Files, LinkOption, Path}
+
+object AstPrinter {
+
+  /** default indentation width of 4 spaces */
+  private val Indent = 4
+
+  /** default maximum output width of 80 chars */
+  private val Width = 80
+
+  /**
+    * The extension of intermediate flix files.
+    */
+  val IrExtension: String = "flixir"
+
+  /**
+    * Prints `ast` to `build/ast/<phase>.<AstPrinter.IrExtension>` if
+    * `flix.options.xprintasts` contains `phase`.
+    *
+    * @param ast    call-by-name to avoid premature computation
+    */
+  def printAst(phase: String, ast: => DocAst.Program)(implicit flix: Flix): Unit = {
```

Comment:
- Drop the by name parameter.
- Split into two functions: one for formatting and one for writing to a file.